pkg/kmm/containers_test.go

```diff
@@ -0,0 +1,230 @@
+
+	for _, testCase := range testCases {
+		testModuleLoaderContainerBuilder := NewModLoaderContainerBuilder(testCase.name)
+		assert.Equal(t, testCase.expectedError, testModuleLoaderContainerBuilder.errorMsg)
+		assert.NotNil(t, testModuleLoaderContainerBuilder.definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testModuleLoaderContainerBuilder.definition.Modprobe.ModuleName)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithModprobeSpec(t *testing.T) {
+	testCases := []struct {
+		dirName            string
+		fwPath             string
+		parameters         []string
+		args               []string
+		rawargs            []string
+		moduleLoadingOrder []string
+	}{
+		{
+			dirName:            "",
+			fwPath:             "",
+			parameters:         nil,
+			moduleLoadingOrder: nil,
+			args:               nil,
+			rawargs:            nil,
+		},
+		{
+			dirName:            "test",
+			fwPath:             "test",
+			parameters:         []string{"one", "two"},
+			moduleLoadingOrder: []string{"one", "two"},
+			args:               []string{},
+			rawargs:            []string{},
+		},
+		{
+			dirName:            "test",
+			fwPath:             "test",
+			parameters:         []string{"one", "two"},
+			moduleLoadingOrder: []string{"one", "two"},
+			args:               []string{"arg"},
+			rawargs:            []string{},
+		},
+		{
+			dirName:            "test",
+			fwPath:             "test",
+			parameters:         []string{"one", "two"},
+			moduleLoadingOrder: []string{"one", "two"},
+			args:               []string{},
+			rawargs:            []string{"arg"},
+		},
+		{
+			dirName:            "test",
+			fwPath:             "test",
+			parameters:         []string{"one", "two"},
+			moduleLoadingOrder: []string{"one", "two"},
+			args:               []string{"arg"},
+			rawargs:            []string{"rawarg1", "rawargs2"},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := NewModLoaderContainerBuilder("test")
+		testBuilder.WithModprobeSpec(testCase.dirName, testCase.fwPath,
+			testCase.parameters, testCase.args, testCase.rawargs, testCase.moduleLoadingOrder)
+
+		assert.Equal(t, testCase.dirName, testBuilder.definition.Modprobe.DirName)
+		assert.Equal(t, testCase.fwPath, testBuilder.definition.Modprobe.FirmwarePath)
+		assert.Equal(t, testCase.parameters, testBuilder.definition.Modprobe.Parameters)
+		assert.Equal(t, testCase.moduleLoadingOrder, testBuilder.definition.Modprobe.ModulesLoadingOrder)
+
+		if len(testCase.args) > 0 {
+			assert.Equal(t, testCase.args, testBuilder.definition.Modprobe.Args.Load)
+		}
+
+		if len(testCase.rawargs) > 0 {
+			assert.Equal(t, testCase.rawargs, testBuilder.definition.Modprobe.RawArgs.Load)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithImagePullPolicy(t *testing.T) {
+	testCases := []struct {
+		imagePolicy    string
+		expectedPolicy string
+		expectedError  string
+	}{
+		{
+			imagePolicy:    "",
+			expectedPolicy: "IfNotPresent",
```

Comment:
Are IfNotPresent, Never and Always only valid values? Can we add them to containers.go as valid list and add verification ?

Reply:
It seems that I was wrongfully using a bad import. Fixed it and updated tests.